dex/pair/src/fee.rs

```diff
@@ -82,128 +82,86 @@ pub trait FeeModule:
                 second_token: first_token,
             };
             is_removed = self.trusted_swap_pair().remove(&token_pair_reversed) != None;
-            require!(is_removed, "Pair does not exist in trusted pair map");
+            assert!(self, is_removed, ERROR_PAIR_NOT_TRUSTED);
         }
-        Ok(())
     }
 
-    fn reinject(&self, token: &TokenIdentifier, amount: &BigUint) {
-        let mut reserve = self.pair_reserve(token).get();
-        reserve += amount;
-        self.pair_reserve(token).set(&reserve);
-    }
-
-    fn send_fee(&self, fee_token: &TokenIdentifier, fee_amount: &BigUint) {
-        if fee_amount == &0 {
+    fn send_fee(
+        &self,
+        context: &mut dyn Context<Self::Api>,
+        fee_token: &TokenIdentifier,
+        fee_amount: &BigUint,
+    ) {
+        if fee_amount == &0u64 {
             return;
         }
 
         let slices = self.destination_map().len() as u64;
-        if slices == 0 {
-            self.reinject(fee_token, fee_amount);
+        if slices == 0u64 {
             return;
         }
```

Comment:
`if self.destination_map().is_empty() { return; }` is a bit cleaner.

Reply:
well, i use slice below, so i dont see a reason why using .is_empty() and .len() instead of just .len()